### ISSUE TITLE
Linux: Follow the XDG Base Directory Specification

### ISSUE THREAD

**Issue by ruby0b** [ArchipelagoMW/Archipelago#4337]
### What feature would you like to see?

When using the AppImage on Linux, Archipelago creates `~/Archipelago` to dump its files into:
https://github.com/ArchipelagoMW/Archipelago/blob/c9625e1b35c36866ee727128fe73808580f37145/Utils.py#L154-L156
This behavior is annoying if you like to keep a clean home directory.
The [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/latest/) exists for this purpose and is widely followed on Linux nowadays.

Since Archipelago doesn't separate its files into clearly separatable directories, the recommended "catch-all" for cases like this is using `$XDG_STATE_HOME` (defaulting to `~/.local/state`).

Therefore I propose using `$XDG_STATE_HOME/Archipelago` as the default Archipelago directory (moving `~/Archipelago` to it if it exists):
```py
    elif sys.platform.startswith('linux'): 
        xdg_state_home = os.getenv('XDG_STATE_HOME', os.path.expanduser('~/.local/state'))
        home_path.cached_path = xdg_state_home + '/Archipelago'
        if not os.path.isdir(home_path.cached_path):
            legacy_home_path = os.path.expanduser('~/Archipelago')
            if os.path.isdir(legacy_home_path):
                os.renames(legacy_home_path, home_path.cached_path)
            else:
                os.makedirs(home_path.cached_path, 0o700, exist_ok=True) 
```

**Comment by black-sliver** [comment]
The problem is that `~/Archipelago/Players/` folder, which needs to be accessible, would then be in a hidden folder.

We need a solution for that before we can move stuff to a hidden folder.

**Comment by ruby0b** [comment]
The "Browse Files" button in the launcher already opens the correct folder, that should be discoverable enough, right?

I mean speaking as a user it's not really much more intuitive to look for a folder in your home directory when the AppImage you launched is in an entirely different location anyway.

**Comment by black-sliver** [comment]
Oh, yeah, it was late and I forgot that we already solved that.

(Setting up ~/Archipelago predates that button)

So there should be nothing in the way other than work. We should defjnitely include a migration code path, maybe create a symlink if the folder was moved so downgrading doesn't break.

**Comment by ruby0b** [comment]
Creating a symlink when moving sounds like a great idea, also has the benefit of possibly being less confusing to existing users :)

I'll create a PR (though I'm not familiar with the build process).

**Comment by beauxq** [comment]
I'm imagining some users might prefer the `~/Archipelago` folder (and not for it to be a link).
Is it something we can make configurable?

**Comment by black-sliver** [comment]
The config is in that folder :thinking: 
Symlink is basically as good as a real folder for the purpose of quick access, no?
If they want to have it on a different mount point, they could create a symlink at $XDG_WHATEVER/Archipelago to that.

**Comment by ruby0b** [comment]
> I hope I get to this soon, I am wondering if state dir is correct though.
> 
> Looking at the XDG specs, it reads to me like if the data in that folder is important and/or portable (between versions?), it should be in XDG_DATA_HOME rather than XDG_STATE_HOME, which i think would apply to most files we put there.

Yeah from looking at other examples I think you're right, will update the PR.
